ocamltest/run_win32.c

```diff
@@ -175,18 +175,40 @@ static LPVOID prepare_environment(WCHAR **localenv)
   }
   r = env;
   p = process_env;
+  /* Copy process_env to env only if the given names are not in localenv */
   while (*p != L'\0') {
+    wchar_t *pos_eq = wcschr(p, L'=');
+    int copy = 1;
     l = wcslen(p) + 1; /* also count terminating '\0' */
-    memcpy(r, p, l * sizeof(WCHAR));
+    /* Temporarily change the = to \0 for wcscmp */
+    *pos_eq = L'\0';
+    q = localenv;
+    while (*q != NULL) {
```

Comment:
I'd use a `for` loop here.

Reply:
This was in keeping with similar loops already present in the function, but I agree that they look clearer as `for` loops - I changed them all in de11da3.